projects/firestore/build.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -eu
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# build project
+# e.g.
+# ./autogen.sh
+# ./configure
+# make -j$(nproc) all
+
+# build fuzzers
+# e.g.
+# $CXX $CXXFLAGS -std=c++11 -Iinclude \
+#     /path/to/name_of_fuzzer.cc -o $OUT/name_of_fuzzer \
+#     -lFuzzingEngine /path/to/library.a
+
+# Download cmake binaries becaue the cmake version that is installed using
+# `apt-get install cmake` is older than the minimum requirements.
+cd $WORK
+wget https://cmake.org/files/v3.12/cmake-3.12.0-Linux-x86_64.tar.gz
```

Comment:
do this in Dockerfile please (apt-get is preferred, if you can make it work)

Reply:
This is going to be very tricky. We depend on [leveldb](https://github.com/google/leveldb) which [requires cmake 3.9](https://github.com/google/leveldb/blob/master/CMakeLists.txt#L5). I contacted them to downgrade to 3.5 to match the Ubuntu 16.04 version but I have no idea if they are willing to do this or not.

So, our options are:
1. Download cmake binaries as we do here.
2. Download cmake source and build it ourselves.
3. Patch leveldb to modify its cmake minimum requirements to 3.5 but this can introduce problems later on.
4. Use a thirdparty ppa repository to install a more recent version of cmake, which is not ideal.

None of these solutions is ideal, so I decided to go with 1 as it is straightforward.